src/games/ecards.py

```diff
@@ -210,7 +214,9 @@ async def execute_round(self, leader: str, prompt: str, hands: dict):
                 )
 
                 # update scoreboard
-                self._add_score(winning_pid, 1)
+                return self._add_score(winning_pid, 1)
+            else:
+                return
```

Comment:
Put a log statement here so we have some record of whether this code is ever being reached. Maybe 
```py
self.logging.warning(f"{winning_card} not in {cards_played}")
```
or something similarly verbose, so we have some way to work out what went wrong.